SU2_CFD/src/solvers/CDiscAdjSolver.cpp

```diff
@@ -164,6 +164,9 @@ void CDiscAdjSolver::RegisterSolution(CGeometry *geometry, CConfig *config) {
 
   direct_solver->RegisterSolutionExtra(true, config);
 
+  /*--- Register quantities that are no solver variables but further inputs/outputs of the (outer) iteration. ---*/
+  direct_solver->RegisterComplementary(true, config);
```

Comment:
Can't you put this in solution extra?

Reply:
Yes! Actually that's what I did originally but RegisterSolutionExtra is also used in the constructor to determine an extra allocation of adjoint variables. I let you decide, I've tested both solutions..

Reply:
Yep move it there please, we already have quite a variety of things to register, solution, extra, variables, geometry, :rofl: